### PR TITLE
Fix Stubborn Checkboxes on Stream

### PR DESCRIPTION
**Pull request by montlebalm**
#### What's this PR do?

This PR fixes the uncooperative checkboxes on the Stream page. Previously, the "check all" box would stay checked even if one or more of the child checkboxes became unchecked.

As part of the refactor, we also added new methods to `SelectedGroupStore` including `add()`, `prune()`, `allSelected()`, `anySelected()`, and `multiSelected()`. In addition we removed: `allSelected`, `anySelected`, `multiSelected`, `refresh()`.

This PR also includes ES5 and ES6 shims, so UI code will work in older browsers.

This fixes "Unticking a single group should untick select all" on issue #1585.
#### Where should the reviewer start?

Refactor: `src/sentry/static/sentry/app/stores/selectedGroupStore.jsx`
Tests: `tests/js/spec/stores/selectedGroupStore.spec.js`
#### How should this be manually tested?

Run `make test-js`
#### Any background context you want to provide?

We decided to move some state from the `SelectedGroupStore` into methods that calculate their value as needed.

We also added some depth to the `tests/js/spec/` directory to mirror the hierarchy of `static/app/`.
#### What gif best describes this PR or how it makes you feel?

magics

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
   "dependencies": {
     "bootstrap": "^3.3.4",
     "crypto-js": "^3.1.2-5",
+    "es5-shim": "^4.1.7",
```
Comment (dcramer) [inline package.json]: is this related to the us using Set() etc ?

**Comment by dcramer** [comment]
this lgtm

File: package.json (hunk 1 above)
Comment (benvinegar) [inline package.json]: Not sure I understand why es5-shim is necessary ... I thought Sentry targeted ES5-compatible browsers.

File: package.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
...
+    "es6-shim": "^0.32.2",
```
Comment (benvinegar) [inline package.json]: Is es6-shim necessary when everything is being run through Babel?

File: package.json (hunk 1 above)
Comment (tracehelms) [inline package.json]: Yes, `Set()` definitely had a part in this.

Comment (montlebalm) [inline package.json]: The ES6 shim adds objects like `Set`. The shim for ES5 polyfills methods like `Array.map` and `Object.keys`.

- ES5 shim: https://github.com/es-shims/es5-shim
- ES6 shim: https://github.com/paulmillr/es6-shim/

Comment (benvinegar) [inline package.json]: Re: ES6 shim - we should probably use the Babel polyfill, since it only
polyfills the things not already covered by Babel.

http://babeljs.io/docs/usage/polyfill/

Re: ES5 - all of those methods are supported by IE9 and up (
http://kangax.github.io/compat-table/es5/).

Comment (tracehelms) [inline package.json]: It looks like the ES6 shim is not necessary because we're using `babel-loader`. I'm taking out `es5-shim` and `es6-shim`. Tested in [Firefox 12 which doesn't support](https://developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/ECMAScript_6_support_in_Mozilla) `Set()` and `Map()`.
